tests/test_string.f90

```diff
@@ -0,0 +1,46 @@
+program main
+  use string, only : get_double, get_real, get_integer, str
+  use iso_fortran_env, only : real32, real64, int32
+  implicit none
+
+  ! Enumerate the test-result array indices:
+  enum, bind(C)
+    enumerator ::  get_double_, get_real_, get_integer_, str_d_, str_r_, str_i_
+  end enum
+  logical :: test_passed(get_double_:str_i_)=.false.
+
+  ! Test data
+  real(real64), parameter :: real64_datum=0.3_real64
+  real(real32), parameter :: real32_datum=0.3_real32
+  integer,      parameter :: integer_datum=1234567
+
+  ! Construct expected results
+  character(len=32) :: real64_string, real32_string, integer_string
+
+  write(real64_string,*) real64_datum
+  write(real32_string,*) real32_datum
+  write(integer_string,*) integer_datum
+
+  ! Test each string conversion function
+  if ( get_double(real64_string)   == real64_datum         ) test_passed(get_double_) = .true.
```

Comment:
I assume these tests pass at the moment, but are there potential issues with precision?  Should it only check that the difference between values is less than some small value rather then requiring them to be exactly equal?